lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,18 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(@parameters[:name].value) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      file_resources = catalog.resources.reject { |resource| resource.type != Puppet::Type.type(:file) }
+      children_file_resources = file_resources.select { |resource| resource.parameters[:path].value =~ %r(#{@parameters[:name].value}/.) }
```

Comment:
I think you can simplify `resource.parameters[:path].value` to `resource[:path]`. Calling `resource.parameters[:path]` will return the instance of `Puppet::Type::File::ParameterPath` (created via the `newparam(:path)` DSL method). 

Similarly, I think you can simplify `@parameters[:name].value` to `self[:path]`. The autorequire block is evaluated in the context of the specific `mount` resource, so `self` here is an instance of `Puppet::Type::Mount`.